### ISSUE TITLE
Deal with empty target site alleles appropriately

### ISSUE THREAD

**Issue by Lioscro** [YosefLab/Cassiopeia#139]
Some target sites end up getting an empty string `''` as their allele. Still not entirely sure where/why this is happening, but the pipeline should deal with these more appropriately.
Currently, if a table has such empty-string-target site entries, reading the table with `pd.read_csv` causes these entries to be replaced with a floating-point `NaN`, which causes problems when we try to use the target site indels to construct a string. Specifically, if we try to join the target sites into a single allele string, as so
```python
molecule_table[['r1', 'r2', 'r3']].apply(lambda x: '_'.join(x))
```
this will cause an error because all the arguments to `join` must be strings, but `NaN` is a float.

One (hacky?) way to get around this is to simply provide `na_filter=False` to `pd.read_csv`, which reads the empty strings literally. But it is unclear if this is the behavior we want. Or do we want to simply filter out all UMIs with missing target site indels?

Here is an example of a UMI that has a missing target site indel.
```
AAAAAACCGTCGTA_GTGGTCGGG_1      777.0   110M1D52M1D55M  0.0     0.0     AATCCAGCTAGCTGTGCAGCCGTGAGTCTCTGATATTCAACTGCAGTAATGCTACCTAGTACTCACGCTTTCCAAGTGCTTGGCGTCGCATCACGGTCCTTTGTACGCCGAAAATCGCCAGACAACTAAGCTACGGCACGCTGCCATGTTGGGTCATACCCAAAATCAGGTTACTCCTGGGCCGCACAAGTCATGGAGAAGTCGAGACTATTAATTCATGATTCCCAATCTGGACTTACTACATGTATACCCC   GTGGTCGGG       [111:1D][164:1D]AAAAAACCGTCGTA  CGTGAGTCTCTGAT  [111:1D]        [164:1D]                1.0
```

After discussing with @mattjones315, this behavior is mainly due to the low sequencing quality of this dataset, compounded by technical artifacts in local alignment (no alignment for the particular target site)

**Comment by Lioscro** [comment]
For the previous example, you could make the argument that there are a lot of mismatches at the third site, which is why the pipeline is having a hard time accurately calling the third indel (and hence the empty string). However, here is a more egregious example.
```
ACTCGGTCGGCGCA_GTCCTACCG_39     1020.0  218M    0.0     0.0     AATCCAGCTAGCTGTGCAGCCGCTATGGGGGGAAATTCAACTGCAGTAATGCTACCTCGTACTCACGCTTTCCAAGTGCTTGGCGTCGCATCTCGGTCCTTTGTACGCCGAAAAATGGCCTGACAACTAAGCTACGGCACGCTGCCATGTTGGGTCATAACGATATCTCTGGTTCATCCGTGACCGAACATGTCATGGAGTAGCAGGAGCTATTAATTAAAACGGTAGGACTGCGTCTCTCGGGAACGCTGAA   GTCCTACCG       [None][None]    ACTCGGTCGGCGCA  CGCTATGGGGGGAA  [None]  [None]          39.0
```

And alignment
```
seq      AATCCAGCTAGCTGTGCAGCCGCTATGGGGGGAAATTCAACTGCAGTAATGCTACCTCGT	60
ref      AATCCAGCTAGCTGTGCAGCNNNNNNNNNNNNNNATTCAACTGCAGTAATGCTACCTCGT	60
         ********************              **************************

seq      ACTCACGCTTTCCAAGTGCTTGGCGTCGCATCTCGGTCCTTTGTACGCCGAAAAATGGCC	120
ref      ACTCACGCTTTCCAAGTGCTTGGCGTCGCATCTCGGTCCTTTGTACGCCGAAAAATGGCC	120
         ************************************************************

seq      TGACAACTAAGCTACGGCACGCTGCCATGTTGGGTCATAACGATATCTCTGGTTCATCCG	180
ref      TGACAACTAAGCTACGGCACGCTGCCATGTTGGGTCATAACGATATCTCTGGTTCATCCG	180
         ************************************************************

seq      TGACCGAACATGTCATGGAGTAGCAGGAGCTATTAATTAAAACGGTAGGACTGCGTCTCT	240
ref      TGACCGAACATGTCATGGAGTAGCAGGAGCTATTAATTCGCGGAGGACAATGCGGTTCGT	240
         **************************************      * *  *    **   *

seq      CGGGAACGCTGAA-----------------------------------------------	253
ref      AGTCACTGTCTTCCGCAATCGTCCATCGCTCCTGCAGGTGGCCTAGAGGGCCCGTTTAAA	300
          *  *  *                                                    

seq      ------------------------------------------------------------	253
ref      CCCGCTGATCAGCCTCGACTGTGCCTTCTAGTTGCCAGCCATCTGTTGTTTGCCCCTCCC	360
                                                                     

seq      ------------------------------------------------------------	253
ref      CCGTGCCTTCCTTGACCCTGGAAGGTGCCACTCCCACTGTCCTTTCCTAATAAAATGAGG	420
                                                                     

seq      ------------------------------------------------------------	253
ref      AAATTGCATCGCATTGTCTGAGTAGGTGTCATTCTATTCTGGGGGGTGGGGTGGGGCAGG	480
                                                                     

seq      ------------------------------------------------------------	253
ref      ACAGCAAGGGGGAGGATTGGGAAGACAATAGCAGGCATGCTGGGGATGCGGTGGGCTCTA	540
                                                                     

seq      ------------------------------------------------------------	253
ref      TGGTCTAGAGCGGGCCCGGTACTAACCAAACTGGATCTCTGCTGTCCCTGTAATAAACCC	600
                                                                     

seq      ------------------------------------------------------------	253
ref      GAAAATTTTGAATTTTTGTAATTTGTTTTTGTAATTCTTTAGTTTGTATGTCTGTTGCTA	660
                                                                     

seq      ------------------------------------------------------------	253
ref      TTATGTCTACTATTCTTTCCCCTGCACTGTACCCCCCAATCCCCCCTTTTCTTTTAAAAT	720
                                                                     

seq      ------------------------------	253
ref      TGTGGATGAATACTGCCATTTGTCTGCAGA	750
```

**Comment by Lioscro** [comment]
One more
```
CTTCAAGATTGGGC_ACCAGTAAG_70     976.0   110M2D107M      0.0     0.0     AATCCAGCTAGCTGTGCAGCTAACCAAGCCTACAATTCAACTGCAGTAATGCTACCTCGTACTCACGCTTTCCAAGTGCTTGGCGTCGCATCTCGGTCCTTTGTACGCCGAAATGGCCTGACAACTAAGCTACGGCACGCTGCCATGTTGGGTAATAACGATATCTCTGGTTCATCCGTGACCGAACATGTCATGGAGTAGCAGGAGCTATTAATGCTATTAACGGAGCTATTAACGGAGCTATTAACGGAGC   ACCAGTAAG       [111:2D][None] CTTCAAGATTGGGC   TAACCAAGCCTACA  [111:2D]        [None]          70.0
```

```
seq      AATCCAGCTAGCTGTGCAGCTAACCAAGCCTACAATTCAACTGCAGTAATGCTACCTCGT	60
ref      AATCCAGCTAGCTGTGCAGCNNNNNNNNNNNNNNATTCAACTGCAGTAATGCTACCTCGT	60
         ********************              **************************

seq      ACTCACGCTTTCCAAGTGCTTGGCGTCGCATCTCGGTCCTTTGTACGCCGA--AATGGCC	118
ref      ACTCACGCTTTCCAAGTGCTTGGCGTCGCATCTCGGTCCTTTGTACGCCGAAAAATGGCC	120
         ***************************************************  *******

seq      TGACAACTAAGCTACGGCACGCTGCCATGTTGGGTAATAACGATATCTCTGGTTCATCCG	178
ref      TGACAACTAAGCTACGGCACGCTGCCATGTTGGGTCATAACGATATCTCTGGTTCATCCG	180
         *********************************** ************************

seq      TGACCGAACATGTCATGGAGTAGCAGGAGCTATTAATGCTATTAACGGAGCTATTAACGG	238
ref      TGACCGAACATGTCATGGAGTAGCAGGAGCTATTAATTCGCGGAGGACAATGCGGTTCGT	240
         ************************************* *    *    *        ** 

seq      AGCTATTAACGGAGC---------------------------------------------	253
ref      AGTCACTGTCTTCCGCAATCGTCCATCGCTCCTGCAGGTGGCCTAGAGGGCCCGTTTAAA	300
         **  * *  *                                                  

```

**Comment by mattjones315** [comment]
This should be fixed with the merging of #142, implementing a global alignment option.
